src/Aspire.Hosting/ApplicationModel/EndpointNameAttribute.cs

```diff
@@ -0,0 +1,22 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// Signifies that a parameter represents an endpoint name.
+/// <para>
+/// Decorate <see cref="string"/> parameters representing endpoint names with this attribute, e.g.:
+/// <code lang="csharp">
+/// public static IResourceBuilder&lt;TResource&gt; WithCustomEndpoint&lt;TResource&gt;(this IResourceBuilder&lt;TResource&gt; builder, [EndpointName] string name)
+/// </code>
+/// </para>
+/// </summary>
+/// <remarks>
+/// This API supports analyzers in Aspire.Hosting.Analyzers.
+/// </remarks>
+[AttributeUsage(AttributeTargets.Parameter, AllowMultiple = false)]
+public class EndpointNameAttribute : Attribute, IModelNameParameter
```

Comment:
Can we remove this attribute until it is needed? I'm wondering if we actually even need this attribute, or if the analyzer would just hard-code the ~5 methods that take an EndpointName.

Reply:
hard code, ewww. Could go back to what I had originally which is just a single attribute `ModelNameAttribute` that aligns with the `ModelName` type as now we're only checking for valid model name anyway.

Reply:
This is needed as the analyzer needs to pass a "target" to `ModelName.TryValidate` which becomes part of the message. Originally I had it like `[ModelName("Resource")]` or `ModelName("Endpoint")` but that looked super repetitive and jank so I made it strongly typed.